scripts/fetch_vault_credentials.py

```diff
@@ -0,0 +1,86 @@
+#!/usr/bin/env python3
+"""
+Fetch secrets from Vault for source-ing into the application environment.
+
+This script connects to a Vault instance via its HTTP API and attempts
+to pull sensitive environment variables from a namespace as specified by
+the VAULT_* environment variables.
+"""
+import json
+import os
+import sys
+import urllib.request
+
+
+def main():
+    for env_var in [
+        "VAULT_ROLE_NAMESPACE",
+        "VAULT_ROLE_ID",
+        "VAULT_ROLE_SECRET",
+        "VAULT_ADDRESS",
+        "VAULT_NAMESPACE",
+        "VAULT_KV_STORE",
+        "VAULT_SECRET",
+    ]:
+        if env_var not in os.environ:
+            raise OSError(f"The environment variable {env_var} must be set.")
+
+    VAULT_ROLE_NAMESPACE = os.getenv("VAULT_ROLE_NAMESPACE")
+    VAULT_ROLE_ID = os.getenv("VAULT_ROLE_ID")
+    VAULT_ROLE_SECRET = os.getenv("VAULT_ROLE_SECRET")
+    VAULT_ADDRESS = os.getenv("VAULT_ADDRESS")
+    VAULT_NAMESPACE = os.getenv("VAULT_NAMESPACE")
+    VAULT_KV_STORE = os.getenv("VAULT_KV_STORE")
+    VAULT_SECRET = os.getenv("VAULT_SECRET")
+
+    vault_token = fetch_token(
+        VAULT_ADDRESS, VAULT_ROLE_NAMESPACE, VAULT_ROLE_ID, VAULT_ROLE_SECRET
+    )
+    secret = fetch_secret(
+        VAULT_ADDRESS, vault_token, VAULT_NAMESPACE, VAULT_KV_STORE, VAULT_SECRET
+    )
+
+    if not sys.stdout.isatty():
+        print(
+            "export " + " ".join(f"'{key}'='{value}'" for key, value in secret.items())
+        )
```

Comment:
Do we want to change this to:

```python
if sys.stdout.isatty():
    sys.exit(1)
```
?

Otherwise, we may be surprised by why it's not outputting anything.